src/mscorlib/src/System/String.cs

```diff
@@ -2577,6 +2581,150 @@ public static int Compare(String strA, String strB, StringComparison comparisonT
             }                        
         }
 
+#if FEATURE_CORECLR
+        [Pure]
+        private unsafe Boolean StartsWithOrdinal(String startsWith)
+        {
+            var byteCount = startsWith.Length << 1;
+            // value.Length verified to be less than or equal to this.Length by calling function
+            if (byteCount > 512)
```

Comment:
If I am reading your data correctly, the fcall is always slower. Why to call it for larger strings?